addons/mail/static/src/core/web/mail_composer_attachment_list.js

```diff
@@ -20,7 +20,9 @@ export class MailComposerAttachmentList extends Many2ManyBinaryField {
     async onFileRemove(fileId) {
         super.onFileRemove(fileId);
         const attachment = this.mailStore.Attachment.get(fileId);
-        await this.attachmentUploadService.unlink(attachment);
+        if (attachment) {
+            await this.attachmentUploadService.unlink(attachment);
```

Comment:
Thanks for the fix, we should use insert as in future versions, this will avoid the issue of not finding the attachment. 

```suggestion
            await this.attachmentUploadService.insert(attachment);
```

Reply:
My bad, it is the line above what we should modify: `const attachment = this.mailStore.Attachment.insert(fileId);`